### ISSUE TITLE
Variable with functions union type compiles even though it does not satisfy it

### ISSUE THREAD

**Issue by Gabrn** [Microsoft/TypeScript#28204]
**TypeScript Version:**  3.2.0-dev.20181027

**Search Terms:**
functions union
**Code**

```ts
type T = (x: string) => string;
type Q = () => number;
type S = T | Q;
const x: S = () => "hello"; // should be error as there is no option for type ()=>string
```

**Expected behavior:**
Should produce errors for line 4: "hello" is not a number
**Actual behavior:**
Produces no errors for line 4.
**Playground Link:**
https://www.typescriptlang.org/play/#src=type%20T%20%3D%20(x%3A%20string)%20%3D%3E%20string%3B%0D%0Atype%20Q%20%3D%20()%20%3D%3E%20number%3B%0D%0Atype%20S%20%3D%20T%20%7C%20Q%3B%0D%0Aconst%20x%3A%20S%20%3D%20()%20%3D%3E%20%22hello%22%3B%0D%0A

**Related Issues:** 
https://github.com/Microsoft/TypeScript/issues/17863

**Comment by fatcerberus** [comment]
It actually has nothing to do with the union, the function is directly assignable to `T`:
```js
type T = (x: string) => string;
const x: T = () => "hello"; // no error here either
```
which actually makes sense to me because, since the return types match, the function that takes zero parameters can readily be used in place of one that takes 1.

**Comment by Gabrn** [comment]
Ok that answers it then :)
